vispy/gloo/buffer.py

```diff
@@ -58,6 +73,66 @@ def nbytes(self):
 
         return self._nbytes
 
+    @property
+    def use_cpu(self):
+        """ Whether a local copy of data is kept on CPU memory """
+
+        return self._use_cpu
+
+    @property
+    def immediate_upload(self):
+        """ Whether changes to the local CPU memory must be immediately sent to GPU """
+
+        return self._immediate_upload
+
+    @property
+    def cpu_data(self):
+        """ Local copy of data kept on CPU memory (CPUData object) """
+
+        if not self._use_cpu:
+            raise RuntimeError("Buffer has no local CPU buffer.")
+
+        return self._cpu_data
+
+    @cpu_data.setter
+    def cpu_data(self, data):
+        """ Local copy of data kept on CPU memory (CPUData object)
+
+        Parameters
+        ----------
+        data : ndarray | None
+            Data to be uploaded (and also kept on CPU memory).
+        """
+
+        if not self._use_cpu:
+            raise RuntimeError("Buffer has no local CPU buffer.")
+
+        self.set_data(data)
+
+    @property
+    def cpu_data_bytes(self):
+        """ Local CPU data, viewed as a mere bytes vector """
+
+        if not self._use_cpu:
+            raise RuntimeError("Buffer has no local CPU buffer.")
+
+        return self._cpu_data.view(np.uint8).ravel()
+
+    def upload_GPU(self):
```

Comment:
Rename to `upload_gpu` if keeping with this name. I'm not sure I have a preference. `upload_to_gpu` or `upload_from_local` or `upload_from_cpu` even. @kmuehlbauer Other ideas?

Reply:
No preferences, but @larsoner might know, if there are some guidelines.

Reply:
no ideas

Reply:
i made a new commit with `upload_to_gpu`, and docstring correction